src/Stratis.Features.FederatedPeg/SourceChain/Result.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Stratis.Features.FederatedPeg.SourceChain
+{
+    /// <summary>
+    /// A generic result type.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to return if the result was successful.</typeparam>
+    public struct Result<T>
```

Comment:
Okay this seems like an overengineer now haha, don't we use one of these elsewhere? Or borrow one from some package?

Reply:
We do but I didn't want to add another dependency to FS.